### ISSUE TITLE
Upgrading OtripleS to the new Standard

### ISSUE THREAD

**Issue by hassanhabib** [hassanhabib/OtripleS#437]
Hi everyone,
Sorry for the long delay.
I have been taking the time to fundamentally change the way we do our validations in our Foundation services. And now after I have tested and devised a new Standard for doing validations I need volunteers to help me upgrade ALL our foundation services in OtripleS with the new Standard.
Please hit the like button on this message so I know you are interested.

Here's a video about the new Standard:
https://www.youtube.com/watch?v=Wtpxb7yPQP0

Here's the documentation:
https://github.com/hassanhabib/The-Standard/blob/master/2.%20Services/2.1%20Foundations/2.1%20Foundations.md#2131-validation

here's a PR for Student Validations Standard Upgrade:
#438 (For Service Upgrade)
https://github.com/hassanhabib/OtripleS/commit/c453a9cc11a0f8e23ab71d89b4cfe620b8932994 (For Controller Upgrade)

Ideally, we know our process of upgrade succeeded iff:
- Your happy path still works (Acceptance Tests)
- You can see accumulative validation errors with problem details in your controller when you post a default JSON object as follows:

![image](https://user-images.githubusercontent.com/1453985/132973378-854dcbb7-835e-400b-948d-841fc81a3148.png)

**Comment by viralpandya** [comment]
Please let me know when can I work on it

**Comment by eriadhami** [comment]
Same for me.

**Comment by hassanhabib** [comment]
@viralpandya && @eriadhami I will publish assignments tomorrow night just to make sure I get everyone's vote. Thank you for your contributions.

**Comment by hassanhabib** [comment]
Okay friends, here's our assignments for this week - Let's take it from the top. And do it in order of CRUD operations.
For this week:
@viralpandya  you will take on updating validations and controllers to the new Standard for `AssignmentAttachment` model.
@eriadhami  you will take on the next model, which is the `Assignment` model.

You can use the Student service for `Register` validations as a guidance in addition to the documentation on The Standard.
Your deadlines are the end of the week. Thank you both. <3 

Let me know if you have any issues.

**Comment by stasnowak** [comment]
@hassanhabib Sorry for being late to the party, i did not check after the great video in github itself. Let me know if i can join on a model and try to implement it based on new standard.

greetings

**Comment by hassanhabib** [comment]
@stasnowak take on `CalendarEnties` service upgrade

**Comment by hassanhabib** [comment]
@eriadhami && @stasnowak  are you guys ready for another round this week?

**Comment by eriadhami** [comment]
@hassanhabib yes, i will be happy

**Comment by hassanhabib** [comment]
@eriadhami  okay you take `Calendars`
I will take `Classrooms`

Thank you.

**Comment by stasnowak** [comment]
I'm ready as well ;)

**Comment by hassanhabib** [comment]
@stasnowak Awesome - take the `Contacts`

**Comment by bilalmehrban** [comment]
@hassanhabib which one I should go for?

**Comment by hassanhabib** [comment]
@bilalmehrban you will take `Courses` and I will take `CoursesAttachment`

**Comment by hassanhabib** [comment]
@bilalmehrban , @stasnowak and @eriadhami  please reference this issue number in our PRs when you create one.

**Comment by mtoncic** [comment]
Hi there, I would also like to contribute. 
I guess ExamAttachment would be next.

**Comment by hassanhabib** [comment]
@mtoncic awesome - if you fork `OtripleS` and open up a PR against this repo you wouldn't need to worry about the permissions.

**Comment by mtoncic** [comment]
I guess next one available would be ExamFees, so If nobody has something against I would pick that one.

**Comment by hassanhabib** [comment]
@stasnowak @eriadhami @bilalmehrban you guys ready for another round?

**Comment by bilalmehrban** [comment]
@hassanhabib please check MS teams, I have sent you a message.

**Comment by ShriHumrudha** [comment]
@hassanhabib I see Attachments, Attendances, CalendarEntryAttachments are not taken in the order of the models. Please count me in and let me know my assignment.

**Comment by eriadhami** [comment]
> @stasnowak @eriadhami @bilalmehrban you guys ready for another round?

@hassanhabib I am sorry bro but I have been busy all week so I could not complete my last task. I will try to catch up at the end of the week.

**Comment by hassanhabib** [comment]
Okay guys. here's our assignments:
I will take `CalendarEntryAttachment`
@bilalmehrban take the `Calendar`
@stasnowak  take the `Classroom`
@mtoncic  you are taking `ExamFee`
@ShriHumrudha take `Course`

I will check with you at the end of the week. thank you all.

**Comment by mtoncic** [comment]
Hi Hassan,

I see on some places other CRUD operations then Create were not considered. Also in your Students example. That might confuse people if they take it as a reference. Just for clarification - All CRUD operations should be tackled in the PR for the taken model?

**Comment by hassanhabib** [comment]
@mtoncic yes. We should apply the new validation Standard to all CRUD operations. Even if it was only for one attribute like a `studentId`. Thanks for asking.

**Comment by hassanhabib** [comment]
@eriadhami take the `Teachers` service
@mtoncic  take the `UserContact` service.

**Comment by emmicourt** [comment]
@hassanhabib  - Can I jump in and help with the next open model? I see `Exam` has not been updated yet.

**Comment by hassanhabib** [comment]
@emmicourt go for it :)
welcome aboard friend :)

**Comment by hassanhabib** [comment]
@devmanzur sure thing. welcome back! :) 
Please make sure all validations on the `StudentService` are following the new validation Standard.

**Comment by devmanzur** [comment]
I'm facing an issue with tests, it could be just my computer, I don't know. dateTimeBroker.GetCurrentDateTime() is returning my local time (GMT +6) instead of UTC time. while DateTimeOffset.UtcNow returns the actual UTC time. so I'm getting an extra exception that should not be happening at all.
